### PR TITLE
feat: rpi generic overlay

### PR DESCRIPTION
**Pull request by frezbo**
Adds the `rpi_generic` overlay.

### CODE REVIEW

File: artifacts/u-boot/pkg.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      SOURCE_DATE_EPOCH: {{ .BUILD_ARG_SOURCE_DATE_EPOCH }}
+    prepare:
+      - |
+        mkdir -p /usr/bin \
+          && ln -sf /toolchain/bin/env /usr/bin/env
+
+        tar xf u-boot.tar.bz2 --strip-components=1
+      # rpi_generic (built with rpi_arm64_defconfig, which "uses the device-tree provided by the firmware")
+      - |
+        # patch -p1 < /pkg/patches/0002-rpi-add-NVMe-to-boot-order.patch
```
Comment (frezbo) [inline artifacts/u-boot/pkg.yaml:28]: TODO: needs newer patches

File: installers/rpi_generic/src/go.mod [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module rpi_generic
+
+go 1.22.0
+
+replace github.com/siderolabs/talos/pkg/machinery v1.7.0-alpha.0 => github.com/frezbo/talos/pkg/machinery v0.14.0-alpha.0.0.20240301145124-b00f3018c270
```
Comment (frezbo) [inline installers/rpi_generic/src/go.mod:5]: TODO: replace once https://github.com/siderolabs/talos/pull/8383 is merged

File: artifacts/u-boot/pkg.yaml (hunk 1 above)
Comment (frezbo) [reply artifacts/u-boot/pkg.yaml]: reverted to the version of u-boot from pkgs

File: Pkgfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+format: v1alpha2
+
+vars:
+  # renovate: datasource=git-refs versioning=git depName=https://github.com/raspberrypi/firmware.git
+  raspberrypi_firmware_ref: dc94391863445ab867782d25ca6ae1e88579df5d
+  raspberrypi_firmware_sha256: a37041044b4d46e324dc03f64455a7c79a38aaf77d0bfc03e6ad7052fc217cd1
+  raspberrypi_firmware_sha512: cf5ed2ce16ee059c9c185ef53d079fb4a59720f1729c270480f2ffb46e5d990936ad58233486143f71d60116fd14e78edb81cd13feec1a787a47650d7eaaf56f
+
+  # renovate: datasource=github-tags extractVersion=^v(?<version>.*)$ depName=u-boot/u-boot
+  uboot_version: 2023.01
```
Comment (smira) [inline Pkgfile:12]: should we use latest, or for now just keep it the way it was?

**Review by smira — APPROVED** [review]

**Pull request merged**